github/repos.go

```diff
@@ -583,11 +583,11 @@ type PullRequestReviewsEnforcementRequest struct {
 func (req PullRequestReviewsEnforcementRequest) MarshalJSON() ([]byte, error) {
```

Comment:
Now that it's no longer neccessary to send empty arrays to GitHub, this entire `MarshalJSON` method and a lot of other related code can be deleted. The end result will be much simpler.

I suggest looking at #703 for more information. That PR was on the right track, we just couldn't review it property at the time because GitHub hadn't updated their documentation back then.